src/templates/releaseNoteLandingPage.js

```diff
@@ -81,43 +81,60 @@ const ReleaseNoteLandingPage = ({ data, pageContext, location }) => {
 
             return (
               <Timeline.Item label={date} key={date}>
-                {posts.map((post) => {
-                  const excerpt = getBestGuessExcerpt(post.mdxAST);
-
-                  return (
-                    <div
-                      key={post.version}
-                      css={css`
-                        margin-bottom: 2rem;
-
-                        &:last-child {
-                          margin-bottom: ${isLast ? 0 : '4rem'};
-                        }
-                      `}
-                    >
-                      <Link
-                        to={post.fields.slug}
+                {posts
+                  .sort((postA, postB) => {
+                    const versionA = parseInt(
+                      postA.frontmatter.version
+                        ?.replace('-beta', '')
+                        ?.replace(/\./g, '')
+                    );
+                    const versionB = parseInt(
+                      postB.frontmatter.version
+                        ?.replace('-beta', '')
+                        ?.replace(/\./g, '')
+                    );
+                    if (!(isNaN(versionA) && isNaN(versionB))) {
+                      return versionB - versionA;
+                    }
+                    return 0;
```

Comment:
Couple things here that I think could help you clean this up just a little bit. I'll go through them piece by piece.

If we are comparing 2 posts without a version, we should avoid sorting them (return value `0`). We can catch that early on and return early. By doing this, you can avoid doing the whole `isNan` dance on line 96.

```js
if (!postA.frontmatter.version || !postB.frontmatter.version) {
  return 0
}
```

2nd of all, I see you're doing `replace` twice here to try and get rid of any unwanted characters in the version name. While what you've got works for now, if somebody came along and added an `-alpha`, this would break. Since what you care about here is comparing 2 integers, I'd just remove any non-digits from the string, which you can do in a single regex:

```js
frontmatter.version.replace(/\D/g, '')
```

Lastly, whenever you use `parseInt`, its a good practice to provide the radix argument so that you don't accidentally parse the string as a base of anything other than 10.

```js
parseInt(..., 10) // <-- forces it to parse in base 10
```

With this, you can simplify to returning what you have on line 97:

```js
return versionB - versionA
```

Reply:
Sweet! thank you, these are all great suggestions :) and yes I knew there was someway to accomplish what I wanted with regex, but somehow regex just doesn't jive with my brain. Thank you and I'll make these changes after I get the swiftype PR up :)